pkg/cloudprovider/cloudprovider.go

```diff
@@ -129,7 +137,17 @@ func (c *CloudProvider) LivenessProbe(req *http.Request) error {
 
 // GetInstanceTypes returns all available InstanceTypes
 func (c *CloudProvider) GetInstanceTypes(ctx context.Context, provisioner *v1alpha5.Provisioner) ([]*cloudprovider.InstanceType, error) {
-	nodeTemplate, err := c.resolveNodeTemplate(ctx, provisioner.Spec.Provider, provisioner.Spec.ProviderRef)
+	var nodeTemplate *v1alpha1.AWSNodeTemplate
+	var err error
+	if provisioner.Spec.ProviderRef != nil {
+		nodeTemplate, err = c.resolveNodeTemplate(ctx, nil, &v1.ObjectReference{
+			APIVersion: provisioner.Spec.ProviderRef.APIVersion,
+			Kind:       provisioner.Spec.ProviderRef.Kind,
+			Name:       provisioner.Spec.ProviderRef.Name,
+		})
+	} else {
+		nodeTemplate, err = c.resolveNodeTemplate(ctx, provisioner.Spec.Provider.Raw, nil)
+	}
 	if err != nil {
```

Comment:
optional: do the standard error pattern in two standard error handling if statements, rather than trying to reuse this handler.